models/user.rb

```diff
@@ -0,0 +1,90 @@
+class User < ActiveRecord::Base
+  has_many :study_assignments
+
+  def self.all_inactive
+    Users.all.select do |user|
+      StudyAssignment.where(user_uuid: user.uuid).any? { |assignment| !assignment.undeleted }
+    end
+  end
+
+  def current_assignments
+    study_assignments.undeleted
+  end
+
+  def assign_to_study(study_uuid, role, sites = [])
+    assignment = study_assignments.find { |assignment| assignment.study_uuid == study_uuid }
+
+    unless assignment.active
+      assignment.active = true
+    end
+
+    if assignment
+      if role.is_a?(Array)
```

Comment:
This type of type checking is an antipattern. Think of other ways we can validate that this `role` doesn't come in as an array, or otherwise handle it more gracefully.